.github/workflows/main.yml

```diff
@@ -466,3 +466,84 @@ jobs:
           path: |-
             ${{ github.workspace }}/aws-cdk/dist/
             ${{ github.workspace }}/aws-cdk/**/dist/
+  push-go-runtime:
+    needs: build
+    if: github.event_name == 'push'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Node 12
+        uses: actions/setup-node@v2.1.4
+        with:
+          node-version: '12'
+      - name: Download Artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: built-tree
+          path: ${{ github.workspace }}/jsii-built-tree
+      - name: Extract Artifact
+        working-directory: ${{ github.workspace }}/jsii-built-tree
+        run: |-
+          echo "::group::Untar Archive"
+          tar Jxvf built-tree.tar.xz
+          echo "::endgroup"
+
+          rm built-tree.tar.xz
+      - name: Get Last Commit Message
+        id: get-last-commit-msg
+        working-directory: ${{ github.workspace }}/jsii-built-tree
+        run: |-
+          message=$(git log -1 --format=%s)
+          echo "::set-output name=result::$message"
+      - uses: actions/cache@v2
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+      - name: Install Dependencies
+        working-directory: ${{ github.workspace }}/jsii-built-tree
+        run: |-
+          yarn install --frozen-lockfile
+      - name: Build Runtime Repo Directory
+        id: build-repo-dir
+        run: |-
+          mkdir ${{ github.workspace }}/jsii-runtime-go
+          repo_dir=${{ github.workspace }}/jsii-runtime-go
+          echo "::set-output name=repodir::$repo_dir"
+      - name: Clone Runtime
+        uses: actions/checkout@v2
+        with:
+          repository: ${{ secrets.JSII_RUNTIME_REPO_NAME }}
+          ref: 'main'
+          token: ${{ secrets.AUTO_APPROVE_GITHUB_TOKEN }}
+          path: ${{ steps.build-repo-dir.outputs.repodir }}
+      - name: Copy Runtime Files
+        working-directory: ${{ github.workspace }}/jsii-built-tree/packages/@jsii/go-runtime
+        run: |-
+          yarn ts-node ./build-tools/build-repo-dir.ts ${{ steps.build-repo-dir.outputs.repodir }}
+      - name: Check for changes
+        id: check-for-changes
+        working-directory: ${{ steps.build-repo-dir.outputs.repodir }}
+        run: |-
+          git fetch --depth=1 --quiet origin main
+          git add .
+          # Check for modifications in tracked files
+          changed=$(git diff --name-only origin/main)
+          if [-z "$changed"]; then
+            echo "::set-output name=result::false"
+          else
+            echo "::set-output name=result::true"
+          fi
+      - name: Commit and Push
+        if: steps.check-for-changes.outputs.result == 'true'
+        working-directory: ${{ steps.build-repo-dir.outputs.repodir }}
+        run: |-
+          git config user.name ${{ secrets.JSII_RUNTIME_REPO_USERNAME }}
```

Comment:
Is that necessary? Could we make this be `AWS CDK Automation`, since the email address is hard-coded?

Reply:
Not necessary, relic from manual git authentication stuff.